src/taoensso/sente.cljc

```diff
@@ -828,11 +829,12 @@
                    :requested-reconnect
                    :downgrading-ws-to-ajax
                    :unexpected}] reason)
-     (if (or (:open? state) (not= reason :unexpected))
-       (assoc state
-         :open? false
-         :last-close {:udt (enc/now-udt) :reason reason})
-       state)))
+     (let [state (dissoc state :next-reconnect)]
```

Comment:
Yep, you're right. I didn't click that this was called under unexpected disconnections. Have moved this down below.